### PR TITLE
User Edits Cart

### PR DESCRIPTION
**Pull request by cheljoh**
Users can now remove items from cart. Users cannot add the same ticket twice.

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -62,6 +62,12 @@ def process(tickets)
     self.update(order_total: total)
   end
 
+  def set_ticket_status
+    tickets.each do |ticket|
+      Ticket.find(ticket.id).update(status: "inactive")
+    end
```
Comment (erinnachen) [inline app/models/order.rb:68]: can we change ticket status: "purchased"

File: spec/features/user/user_edits_cart_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    expect(page).to have_content("seat: #{ticket.seat}")
+
+    within("#ticket-#{ticket.id}") do
+      click_on "remove from cart"
+    end
+
+    expect(page).to_not have_content("seat #{ticket.seat}")
+
+    expect(ticket.status).to eq("active")
+  end
```
Comment (erinnachen) [inline spec/features/user/user_edits_cart_spec.rb:33]: 👍

File: app/controllers/cart_tickets_controller.rb [hunk 3]
```diff
@@ -27,14 +27,9 @@ def destroy
...
-  def update
-    @cart.update(params)
-    redirect_to cart_path
-  end
-
   private
 
-  def find_product(id)
-    Product.find(id)
+  def find_ticket(id)
```
Comment (erinnachen) [inline app/controllers/cart_tickets_controller.rb]: it doesn't feel like this needs to be it's own method

File: spec/features/user/user_can_place_an_order_and_view_past_orders_spec.rb [hunk 4]
```diff
@@ -44,7 +44,7 @@
 
     stripe_iframe = all("iframe[name=stripe_checkout_app]").last
     Capybara.within_frame stripe_iframe do
-      page.execute_script(%Q{ $("input#email").val("bob@example.com"); })
+      page.execute_script(%Q{ $("input#email").val("fodays@example.com"); })
```
Comment (erinnachen) [inline spec/features/user/user_can_place_an_order_and_view_past_orders_spec.rb:47]: hahahaha.

File: app/controllers/cart_tickets_controller.rb (hunk 3 above)
Comment (cheljoh) [inline app/controllers/cart_tickets_controller.rb]: Whoops forgot to take that out!

**Comment by erinnachen** [comment]
Cool. Merging to master!

**Pull request merged**